File: packages/python/peru_spatial_utils/crs/utm_zone_for.py
import math


def utm_zone_for(lng: float) -> int:
    if -84 <= lng < -78:
        return 17
    if -78 <= lng < -72:
        return 18
    if -72 <= lng < -66:
        return 19
    return int(math.floor((lng + 180) / 6)) + 1

File: packages/java/src/com/peruspatial/crs/UtmZoneFor.java
package com.peruspatial.crs;

import com.peruspatial.data.CrsData;

public final class UtmZoneFor {
  private UtmZoneFor() {}

  public static int utmZoneFor(double lng) {
    for (CrsData.Zone z : CrsData.ZONES) {
      if (lng >= z.lngMin() && lng < z.lngMax()) return z.zone();
    }
    return (int) Math.floor((lng + 180) / 6) + 1;
  }
}
